bootcamp-api.API/Services/PetService.cs

```diff
@@ -8,32 +8,46 @@
 using bootcamp_api.Data;
 using bootcamp_api.Exceptions;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Mvc;
+using AutoMapper;
 
 namespace bootcamp_api.Services
 {
     public class PetService: IPetService
     {
 
         private readonly PawssierContext _context;
+        private readonly IMapper _mapper;
 
-        public PetService(PawssierContext context)
+        public PetService(PawssierContext context, IMapper mapper)
         {
             _context = context;
+            _mapper = mapper;
         }
 
-        public Pet[] GetAll()
+        public Dto.PetListItem[] GetAll(ApiVersion version, int user_id)
         {
             var pets = _context.Pets
                 .Include(p => p.Conditions)
                 .Include(p => p.Vaccines)
                 .Include(p => p.Prescriptions)
                 .Include(p => p.PetPhoto)
-                .Include(p => p.VetRecords);
```

Comment:
Do you still need to include all these tables to create a PetListItem? It looks like you only need the counts for each child item instead of reading all of the child tables into memory.
You can craft this query so that it return a count of each child table instead returning the entire table.
https://www.ozkary.com/2015/04/entity-framework-associated-table.html
This query will also transform the results and create a PetListItem so you dont need the mapper after you do this.
vart PetItems = _context.Pets.Select(c => new PetListItem()
               {
                   Name = c.Name,
                   Id = c.Id,
		   Breed = c.Breed,
etc.
                   PrescriptionsCount = c.p.Prescriptions.Count(),
                   VaccinesCount = c.Vaccines.Count(),
etc
               }).ToList();
           }